examples/idea-examples/unsupported-data-sources/src/main/kotlin/org/jetbrains/kotlinx/dataframe/examples/hibernate/main.kt

```diff
@@ -0,0 +1,221 @@
+                email = "alice.wang@example.com",
+                country = "Canada",
+            ),
+        )
+    }
+}
+
+private fun SessionFactory.loadCustomersAsDataFrame(): DataFrame<DfCustomers> {
+    return withSession { session ->
+        val criteriaBuilder: CriteriaBuilder = session.criteriaBuilder
+        val criteriaQuery: CriteriaQuery<CustomersEntity> = criteriaBuilder.createQuery(CustomersEntity::class.java)
+        val root: Root<CustomersEntity> = criteriaQuery.from(CustomersEntity::class.java)
+        criteriaQuery.select(root)
+
+        session.createQuery(criteriaQuery)
+            .list()
+            .map { c ->
+                DfCustomers(
+                    address = c.address,
+                    city = c.city,
+                    company = c.company,
+                    country = c.country,
+                    customerId = c.customerId ?: -1,
+                    email = c.email,
+                    fax = c.fax,
+                    firstName = c.firstName,
+                    lastName = c.lastName,
+                    phone = c.phone,
+                    postalCode = c.postalCode,
+                    state = c.state,
+                    supportRepId = c.supportRepId,
+                )
+            }
+            .toDataFrame()
+    }
+}
+
+/**
+ * **Hibernate + Criteria API:**
+ * - ✅ Database-level aggregation (efficient)
+ * - ✅ Type-safe queries
+ * - ❌ Verbose syntax
+ * - ❌ Limited to SQL-like operations
+ */
+private fun SessionFactory.countCustomersPerCountryWithHibernate() {
+    withSession { session ->
+        val criteriaBuilder: CriteriaBuilder = session.criteriaBuilder
+        val query: CriteriaQuery<Tuple> = criteriaBuilder.createTupleQuery()
+        val root: Root<CustomersEntity> = query.from(CustomersEntity::class.java)
+
+        // Select country and count of customers, grouped by country, ordered by count DESC
+        query.multiselect(
+            root.get<String>("country").alias("country"),
+            criteriaBuilder.count(root.get<Long>("customerId")).alias("customerCount")
+        )
+        query.groupBy(root.get<String>("country"))
+        query.orderBy(
+            criteriaBuilder.desc(criteriaBuilder.count(root.get<Long>("customerId")))
+        )
+
+        val results = session.createQuery(query).list()
+        results.forEach { tuple ->
+            val country = tuple.get("country", String::class.java)
+            val count = tuple.get("customerCount", Long::class.java)
+            println("$country: $count customers")
+        }
+    }
+}
+
+/**
+ * **DataFrame approach: **
+ * - ✅ Rich analytical operations
+ * - ✅ Fluent, readable API
+ * - ✅ Flexible data transformations
+ * - ❌ In-memory processing (less efficient for large datasets)
+ */
+private fun DataFrame<DfCustomers>.analyzeAndPrintResults() {
+    println(size())
+
+    // same operation as Exposed example: customers per country
+    groupBy { country }.count()
```

Comment:
here you do use the generated accessors, but not in `.toCustomersEntity()`. Is that done on purpose?